Source/Other/Extensions/StringExtension.swift

```diff
@@ -41,6 +41,24 @@ extension String {
 		}
 	}
 	
+	// stringByReplacingPercentEscapesUsingEncoding is deprecated
+	// Use -stringByRemovingPercentEncoding
+	// however: per https://developer.apple.com/documentation/foundation/nsstring/1409569-stringbyremovingpercentencoding?language=objc
+	// You must call this method only on strings that you know to be percent-encoded
+	// Generally, removingPercentEncoding fails when the original String contains non-escaped percent symbols
+	// so before we replace stringByReplacingPercentEscapesUsingEncoding all over
+	// we should check the string first
+	var isPercentEncoded: Bool {
+		let decoded = self.removingPercentEncoding
+		
+		if (decoded == nil) || (decoded == self) {
+			return false
+		}
+		else {
+			return true
+		}
```

Comment:
Could turn into one line: `return (decoded != nil) && (decoded != self);`

Reply:
Better to use ´guard let decoded = self.stringByReplacingPercentEncoding else { return false }´
Then ´return self != decoded´

Reply:
Oooo even better!

Reply:
will update